snap/validate.go

```diff
@@ -115,8 +116,17 @@ func ValidateApp(app *AppInfo) error {
 		return fmt.Errorf(`"daemon" field contains invalid value %q`, app.Daemon)
 	}
 
+	// Validate app name
+	if app.Name == "" {
+		return fmt.Errorf("snap app name cannot be empty")
+	}
+
+	if !validAppName.MatchString(app.Name) {
+		return fmt.Errorf("invalid snap app name: %q", app.Name)
```

Comment:
I think it would be worth documenting what the correct name pattern is. I know it is a lot more work (perhaps?) but it turns an error message from a "you have hit a glass wall" to "this way please".